src/signals/incident/components/form/ContainerSelect/Selector/WfsLayer.tsx

```diff
@@ -0,0 +1,90 @@
+import { useMapInstance } from '@amsterdam/react-maps';
+import type { Point, FeatureCollection } from 'geojson';
+import type { GeoJSON as GeoJSONLayer } from 'leaflet';
+import React, { useEffect, useState } from 'react';
+import { featureTolocation } from 'shared/services/map-location';
+import ContainerLayer from './MarkerCluster';
+import { MAX_ZOOM_LEVEL, MIN_ZOOM_LEVEL } from '@amsterdam/arm-core/lib/constants';
+import type { WfsLayerProps } from './types';
+import { NO_DATA } from './types';
+import { fetchWithAbort } from '@amsterdam/arm-core';
+import type { ZoomLevel } from '@amsterdam/arm-core/lib/types';
+
+export const isLayerVisible = (zoom: number, zoomLevel: ZoomLevel) => {
+  const { min, max } = zoomLevel;
+  return zoom <= (min ?? MIN_ZOOM_LEVEL) && zoom >= (max ?? MAX_ZOOM_LEVEL);
+};
+
+const WfsLayer: React.FC<WfsLayerProps> = ({ url, options, zoomLevel }) => {
+  const mapInstance = useMapInstance();
+  const [layerInstance, setLayerInstance] = useState<GeoJSONLayer>();
+  const [bbox, setBbox] = useState('');
+  const [data, setData] = useState<FeatureCollection>(NO_DATA);
+
+  /* istanbul ignore next */
+  useEffect(() => {
```

Comment:
Given that we have a lot of complex hooks for the WfsLayer (2 in `WfsLayer.tsx` and 1 in `Selector.tsx`) we could move these to a separate `WfsLayer.hooks.ts file`. Do you agree?

Reply:
Not sure this is the best choice. I have refactored the components and simplified the logic.